lib/v2/screens/explore_screens/vote_screens/proposals/mappers/proposals_state_mapper.dart

```diff
@@ -28,7 +34,24 @@ class ProposalsStateMapper extends StateMapper {
       // Check if the list needs sort
       List<ProposalModel> newProposals = currentType.isReverse ? List<ProposalModel>.from(filtered.reversed) : filtered;
 
-      return currentState.copyWith(pageState: PageState.success, proposals: newProposals);
+      // Add pass values to proposals by campaing type
+      List<List<SupportLevelModel>> supportLevels = values.whereType<List<SupportLevelModel>>().toList();
+      SupportLevelModel? allianceLevels = supportLevels[0][0];
+      SupportLevelModel? campaingLevels = supportLevels[1][0];
+      SupportLevelModel? milestoneLevels = supportLevels[2][0];
+
+      var updatedProposals = newProposals.map((i) {
+        if (i.campaignType == 'alliance') {
+          i = i.copyWith(allianceLevels.voiceNeeded);
+        } else if (i.campaignType == 'cmp.funding') {
+          i = i.copyWith(campaingLevels.voiceNeeded);
+        } else if (i.campaignType == 'milestone') {
+          i = i.copyWith(milestoneLevels.voiceNeeded);
+        }
+        return i;
+      }).toList();
+
+      return currentState.copyWith(pageState: PageState.success, proposals: updatedProposals);
```

Comment:
@n13 
2 - I add the voice need value in each proposal, that matches the campaign type of each proposal.

*I had to add one more field to the model just to show this value in UI, I still create a view model to not modify this one that is for serialization.

Reply:
Ok... I see ... it seems you would only need to call the chain one time, and could cache those values from thereon out. 

I am not really sure how to best capture that in our mapper structure. 

Does this do the same thing over and over (even if it's just code)? 

If so then there may be a way to avoid doing the same thing over and over...

Reply:
One more think we will also have to show referendums, along with proposals... they are very similar, just minor data differences, I am just adding them to V1

Reply:
> Ok... I see ... it seems you would only need to call the chain one time, and could cache those values from thereon out.

Yes, we could but we need refresh.

> Does this do the same thing over and over (even if it's just code)?

Yes, this mapper is executed just when need fetch the same data (scroll, refresh indicator).

> If so then there may be a way to avoid doing the same thing over and over...

Well , that is the reason for having the mapper to not create other functions to do the same.

Reply:
> One more think we will also have to show referendums, along with proposals... they are very similar, just minor data differences, I am just adding them to V1

Nice, when you finish that feature, tag me in that PR to see how to integrate it in v2 :D